### ISSUE TITLE
Include NumPy BLAS/LAPACK info in client.get_versions()

### ISSUE THREAD

**Issue by jakirkham** [dask/distributed#1827]
At the risk of overloading `client.get_versions()` with info, it would be handy to be able to check the NumPy BLAS/LAPACK linkage in here. This can be really helpful when debugging a slow computation or a very strange segfault that might be BLAS or LAPACK related. One way at this info is `numpy.__config__.show()`, but that might be too heavy for `client.get_versions()`. Open to other ways to include this info if there are suggestions.

**Comment by mrocklin** [comment]
I think it's reasonable to include more things.  It's fairly cheap.  We
might also keep get_versions as it is, but make a larger get_info function
that has a wider scope

**Comment by rbubley** [comment]
Where the task is largely about gathering information from workers, I was wondering if the right approach might be to modify Client.run() to be able to return values (or futures). Deciding what information to harvest from the workers would then be in the control of the clients, not reliant on changes to distributed.

**Comment by mrocklin** [comment]
Yes, that's doable today from user-space and a fine solution.

One reason by get_versions doesn't take this approach (it used to) is that it also gathers information from the scheduler, where we try to avoid depending on pickle.  I suspect that in the future, using pickle may be turned off by default in the scheduler.

**Comment by lalitparate** [comment]
Hi, I am first time contributing to open source. Can I wok on it?

**Comment by jhamman** [comment]
@lalitparate - yes, dask is a community driven open-source project. As such, anyone is welcome to work on anything. Let us know if you need help.

**Comment by HsuanTingLu** [comment]
Are we still aiming to show this worker linkage info in ```client.get_versions()``` ?

> I think it's reasonable to include more things. It's fairly cheap. We might also keep get_versions as it is, but make a larger get_info function that has a wider scope
> […](#)
 
Or should I build ```get_info()```?

**Comment by quasiben** [comment]
As others have commented, adding to `get_versions` seems to be a supported idea.   You might want to look at https://github.com/dask/distributed/pull/3567 as it has some updates to `get_versions` as well as tests

**Comment by HsuanTingLu** [comment]
Sure, thanks.
May I ask what exactly do we want to show in ```get_versions()```?
Since there are lots of possible BLAS/LAPACK library linking options in 
Numpy, (seven currently)
I'm not sure if showing [every build info](https://github.com/numpy/numpy/blob/078ac01a85c4db46e7f148829c2e0d0e0f30c36f/numpy/distutils/misc_util.py#L2327-L2354) presented in ```Numpy.show_config()``` is the best idea.

**Comment by HsuanTingLu** [comment]
Another question is how should we fit the various library linkage info into ```client.get_versions()```?
It seems to me that the current output layout is not meant to present a list of sublists about a package but to show version info alone,
packing stuff like this into ```get_versions()``` for every worker seems suboptimal

```
blas_mkl_info:
  NOT AVAILABLE
blis_info:
  NOT AVAILABLE
openblas_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None)]
blas_opt_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None)]
lapack_mkl_info:
  NOT AVAILABLE
openblas_lapack_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None)]
lapack_opt_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None)]
```

**Comment by quasiben** [comment]
@HsuanTingLu do you have thoughts on a more optimal layout ?

**Comment by HsuanTingLu** [comment]
No, I don't have one, so I'll probably add it anywhere you guys see fit.

Back to the second question, how should the info be fitted into ```client.get_versions()```?
I'm thinking about adding a ```numpy-config``` sublist under ```host```, or maybe somewhere under ```package::numpy```?

**Comment by quasiben** [comment]
I am +1 on package::numpy.  I understand this to mean something like:

```
 'packages': {'numpy': 'blas_opt_info: {}
```

Is that right ?

**Comment by HsuanTingLu** [comment]
Yeah something like this
```'packages': { 'numpy': '1.18.2', 'blas_opt_info: {}, 'lapack_opt_info: {}}```

**Comment by HsuanTingLu** [comment]
@GenevieveBuckley I have a few commits lying around, I think I'll need a few weeks to put them together

**Comment by GenevieveBuckley** [comment]
Sounds great, thank you!
